ouroboros-network/src/Ouroboros/Network/PeerSelection/LedgerPeers.hs

```diff
@@ -63,8 +66,24 @@ data IsLedgerPeer = IsLedgerPeer | IsNotLedgerPeer
 
 newtype NumberOfPeers = NumberOfPeers Word16 deriving Show
 
-newtype LedgerPeersConsensusInterface m = LedgerPeersConsensusInterface {
-      lpGetPeers :: SlotNo -> STM m (Maybe [(PoolStake, NonEmpty RelayAccessPoint)])
+data LedgerPeersConsensusInterface m = LedgerPeersConsensusInterface {
+
+      -- | Return ledger peer information from given slot number.
+      --
+      lpGetPeers :: SlotNo -> STM m (Maybe [(PoolStake, NonEmpty RelayAccessPoint)]),
+
+      -- | Return ledger peer information from a safe slot number.
+      --
+      -- The concrete implementation of this function comes from the consensus
+      -- layer. It plays a crucial role in managing these peers, by distributing
+      -- the load among locally configured root peers to ledger peers.
+      --
+      -- It's important to use this functionality carefully. We should not
+      -- start leveraging ledger peers unless it's absolutely safe. While the
+      -- likelihood of a successful long-range attack by a powerful adversary is
+      -- relatively low, precautions should be taken to avoid becoming a victim.
+      --
+      lpGetPeersAfterSafeSlot :: STM m (Maybe [(PoolStake, NonEmpty RelayAccessPoint)])
```

Comment:
Maybe a better name would be `lpGetRecentPeers`?

Reply:
And then `UseLedgerCloseToTip` -> `UseRecentLedger`?

Reply:
Sure! 😄